src/zenml/integrations/modal/flavors/modal_orchestrator_flavor.py

```diff
@@ -0,0 +1,177 @@
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Modal orchestrator flavor."""
+
+from enum import Enum
+from typing import TYPE_CHECKING, Optional, Type
+
+from zenml.config.base_settings import BaseSettings
+from zenml.orchestrators import BaseOrchestratorConfig, BaseOrchestratorFlavor
+from zenml.utils.secret_utils import SecretField
+
+if TYPE_CHECKING:
+    from zenml.integrations.modal.orchestrators import ModalOrchestrator
+
+MODAL_ORCHESTRATOR_FLAVOR = "modal"
+
+
+class ModalExecutionMode(str, Enum):
+    """Execution modes for the Modal orchestrator.
+
+    Attributes:
+        PIPELINE: Execute entire pipeline in one Modal function (fastest, default).
+        PER_STEP: Execute each step in a separate Modal function (granular
+            control).
+    """
+
+    PIPELINE = "pipeline"
+    PER_STEP = "per_step"
+
+
+class ModalOrchestratorSettings(BaseSettings):
+    """Modal orchestrator settings.
+
+    Attributes:
+        gpu: The type of GPU to use for the pipeline execution (e.g., "T4",
+            "A100").
+            Use ResourceSettings.gpu_count to specify the number of GPUs.
+        region: The region to use for the pipeline execution.
+        cloud: The cloud provider to use for the pipeline execution.
+        environment: The Modal environment to use for the pipeline execution.
+        timeout: Maximum execution time in seconds (default 24h).
+        min_containers: Minimum containers to keep warm (replaces keep_warm).
+        max_containers: Maximum concurrent containers (replaces concurrency_limit).
+        execution_mode: Execution mode - PIPELINE (default, fastest) or PER_STEP (granular control).
+        synchronous: Wait for completion (True) or fire-and-forget (False).
+        app_warming_window_hours: Hours for app name window to enable container reuse.
+            Apps are reused within this time window for efficiency. Default 2 hours.
+    """
+
+    gpu: Optional[str] = None
+    region: Optional[str] = None
+    cloud: Optional[str] = None
+    environment: Optional[str] = None
+    timeout: int = 86400  # 24 hours (Modal's maximum)
+    min_containers: Optional[int] = (
+        1  # Keep 1 container warm for sequential execution
+    )
+    max_containers: Optional[int] = 10  # Allow up to 10 concurrent containers
+    execution_mode: ModalExecutionMode = (
+        ModalExecutionMode.PIPELINE
+    )  # Default to fastest mode
+    synchronous: bool = (
+        True  # Wait for completion (True) or fire-and-forget (False)
+    )
+    app_warming_window_hours: float = (
+        2.0  # Default 2-hour window for app reuse
+    )
```

Comment:
I think we don't state in the docs anywhere that there is a two-hour default set for this warm window. I think we should, esp just as a note for anyone doing dev work on their pipeline, but also maybe that for prod workflows (esp anything automated / triggers etc) you might want to lower this value quite a bit?